app/src/main/java/org/imozerov/streetartview/ui/catalog/ArtListAdapter.java

```diff
@@ -18,10 +21,19 @@
  * Created by imozerov on 05.02.16.
  */
 public class ArtListAdapter extends RecyclerView.Adapter<ArtListAdapter.ViewHolder> {
-    private List<ArtObjectUi> data;
 
-    public ArtListAdapter(@NonNull List<ArtObjectUi> data) {
-        this.data = data;
+    private final Context mContext;
+    private List<ArtObjectUi> mData;
+
+    private DetailOpener mDetailOpener;
+
+    public ArtListAdapter(Context context, @NonNull List<ArtObjectUi> data) {
+        mContext = context;
+        this.mData = data;
+
+        if (context instanceof DetailOpener) {
```

Comment:
Is it really the best way? I have some concerns on re-use of this adapter anywhere else. 
(This comment is not about that this approach is definitely not work. I'm just raising my concerns.